### PR TITLE
[feat] change match during event create.

### PR DESCRIPTION
**Pull request by rado196**
This PR includes:

---

✅ Completed:
- Create nested navigation for `CreateEvent` screen:
  - __the screen opened with `presentation: 'modal'` options and to make navigation inside of modal we need to create nested navigation, otherwise, root navigation will be changed__
- `SelectMatch` screen with match list.
- Move the event store into a separate file to make it available on both screens (`CreateEvent` and `SelectMatch`).
- Added the ability to disable press on the team icon on the `MatchInfo` component.
- Search matches (by home/away team's name).
- Load the next page of matches on the scroll end.
- `Loading`/`NoResult` components condition updated for `FlatList`.
- `BackButton` moved globally and used in the app.

### CODE REVIEW

File: clients/app/src/i18n/locales/en.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ export default {
   'common-username': 'Username',
   'common-tapToChange': 'Tap to Change',
   'common-close': 'Close',
+  'common-back': 'Back',
```
Comment (hay92k) [inline clients/app/src/i18n/locales/en.ts:12]: remove and replace with icon

File: clients/app/src/views/screens/CreateEvent/CreateEventWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (hay92k) [inline clients/app/src/views/screens/CreateEvent/CreateEventWrapper.tsx:1]: no need for Wrapper, merge with navigation

File: clients/app/src/views/screens/CreateEvent/components/BackButtonComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
```
Comment (hay92k) [inline clients/app/src/views/screens/CreateEvent/components/BackButtonComponent.tsx:1]: move to global

File: clients/app/src/views/screens/CreateEvent/navigation.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+const Stack = createNativeStackNavigator();
+
+interface ICreateEventNavigationProps {
+  tabName?: null | string;
+  matchId?: null | ResourceIdentifier;
+}
+
+const currentTabName = 'CreateEvent';
+
+// order is important !!!
```
Comment (hay92k) [inline clients/app/src/views/screens/CreateEvent/navigation.tsx:16]: it does not.

File: clients/app/src/views/screens/CreateEvent/navigation.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import { IScreenType } from './types';
+
+const Stack = createNativeStackNavigator();
+
+interface ICreateEventNavigationProps {
+  tabName?: null | string;
+  matchId?: null | ResourceIdentifier;
+}
+
+const currentTabName = 'CreateEvent';
```
Comment (hay92k) [inline clients/app/src/views/screens/CreateEvent/navigation.tsx:14]: remove this, use useNavigationWithParams hook.

File: clients/app/src/views/screens/CreateEvent/screens/CreateEvent/components/SelectedMatchComponent.tsx [hunk 6]
```diff
@@ -8,13 +8,17 @@ import { InputSection } from 'views/components/base/InputSection';
...
+import { useNavigation } from '@react-navigation/native';
 
 const store = buildMatchesStore();
 
 const SelectedMatchComponent: React.FC<ISelectedMatchProps> = ({
   matchId,
   onRemoveMatchPress,
+  onMatchChange,
 }) => {
+  const navigation = useNavigation<any>();
```
Comment (hay92k) [inline clients/app/src/views/screens/CreateEvent/screens/CreateEvent/components/SelectedMatchComponent.tsx:20]: use useNavigationWithParams hook

File: clients/app/src/views/screens/CreateEvent/screens/SelectMatch/components/SelectMatchComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      keyExtractor={item => `match-section-${item.id}`}
+      renderItem={renderMatch}
+      data={data}
+      onEndReached={onEndReached}
+      onEndReachedThreshold={0.7}
+      horizontal={false}
+      showsVerticalScrollIndicator={false}
+      ListFooterComponent={
+        !loading ? null : (
+          <Box alignItems="center" paddingTop="4" paddingBottom="10">
```
Comment (hay92k) [inline clients/app/src/views/screens/CreateEvent/screens/SelectMatch/components/SelectMatchComponent.tsx:43]: change to Loader Component

File: clients/app/src/views/screens/CreateEvent/screens/SelectMatch/styles.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import { StyleSheet } from 'react-native';
```
Comment (hay92k) [inline clients/app/src/views/screens/CreateEvent/screens/SelectMatch/styles.ts:1]: move to bottom of file.

File: clients/app/src/views/screens/Home/components/AddActionSheet.tsx [hunk 9]
```diff
@@ -13,9 +13,9 @@ const AddActionSheet: React.FC<IAddActionSheetProps> = ({ isOpen, onClose }) =>
   const { colors } = useTheme();
 
   const closeAndNavigateTo = React.useCallback(
-    screen => {
+    (screen: any, params?: any) => {
```
Comment (hay92k) [inline clients/app/src/views/screens/Home/components/AddActionSheet.tsx:16]: don't clear why we need this

**Review by hay92k — CHANGES_REQUESTED** [review]

File: clients/app/src/i18n/locales/en.ts (hunk 1 above)
Comment (rado196) [reply clients/app/src/i18n/locales/en.ts:12]: okay :)

File: clients/app/src/views/screens/CreateEvent/components/BackButtonComponent.tsx (hunk 3 above)
Comment (rado196) [reply clients/app/src/views/screens/CreateEvent/components/BackButtonComponent.tsx:1]: good idea

File: clients/app/src/views/screens/CreateEvent/screens/CreateEvent/components/SelectedMatchComponent.tsx (hunk 6 above)
Comment (rado196) [reply clients/app/src/views/screens/CreateEvent/screens/CreateEvent/components/SelectedMatchComponent.tsx:20]: updated.

File: clients/app/src/views/screens/CreateEvent/screens/SelectMatch/components/SelectMatchComponent.tsx (hunk 7 above)
Comment (rado196) [reply clients/app/src/views/screens/CreateEvent/screens/SelectMatch/components/SelectMatchComponent.tsx:43]: updated

File: clients/app/src/views/screens/CreateEvent/screens/SelectMatch/styles.ts (hunk 8 above)
Comment (rado196) [reply clients/app/src/views/screens/CreateEvent/screens/SelectMatch/styles.ts:1]: done

File: clients/app/src/views/screens/Home/components/AddActionSheet.tsx (hunk 9 above)
Comment (rado196) [reply clients/app/src/views/screens/Home/components/AddActionSheet.tsx:16]: unnecessary params are removed

**Review by hay92k — APPROVED** [review]
lgtm

**Pull request merged**